web-common/src/components/column-profile/ColumnProfileProvider.svelte

```diff
@@ -0,0 +1,66 @@
+<script lang="ts">
+  import {
+    createColumnsProfileData,
+    setColumnsProfileStore,
+  } from "@rilldata/web-common/components/column-profile/columns-profile-data";
+  import { getTableReferences } from "@rilldata/web-common/features/models/utils/get-table-references";
+  import { getMatchingReferencesAndEntries } from "@rilldata/web-common/features/models/workspace/inspector/utils";
+  import {
+    createQueryServiceTableColumns,
+    createRuntimeServiceListCatalogEntries,
+    V1TableColumnsResponse,
+  } from "@rilldata/web-common/runtime-client";
+  import type { V1CatalogEntry } from "@rilldata/web-common/runtime-client/index.js";
+  import { runtime } from "@rilldata/web-common/runtime-client/runtime-store";
+  import type { CreateQueryResult } from "@tanstack/svelte-query";
+
+  export let objectName: string;
+  export let sql = "";
```

Comment:
It'd be a cleaner API if the SQL were fetched within the component, not passed in